src/industry_cmd.cpp

```diff
@@ -420,7 +421,8 @@ static void AddAcceptedCargo_Industry(TileIndex tile, CargoArray &acceptance, Ca
 	MemCpyT(accepts_cargo, itspec->accepts_cargo, lengthof(accepts_cargo));
 	MemCpyT(cargo_acceptance, itspec->acceptance, lengthof(cargo_acceptance));
 
-	if (itspec->special_flags & INDTILE_SPECIAL_ACCEPTS_ALL_CARGO) {
+	/* Non-NewGRF (vanilla) industry tiles accept all cargos. */
+	if (itspec->special_flags & INDTILE_SPECIAL_ACCEPTS_ALL_CARGO || (itspec->grf_prop.grffile == nullptr && indspec->grf_prop.grffile == nullptr)) {
```

Comment:
Bikeshed: I would have put "( )" around the "(a & b)" term.

Reply:
I'd consider pulling the grffile condition into its own variable. Self documenting code!
Relatedly, "accept all cargos" could be clearer - "accept all cargoes of that industry". Also, cargoes.